dandiapi/api/models/metadata.py

```diff
@@ -20,10 +28,10 @@ def published_by(cls, now: datetime.datetime):
             'wasAssociatedWith': [
                 {
                     'id': uuid4().urn,
-                    'identifier': 'RRID:SCR_017571',
-                    'name': 'DANDI API',
+                    **({'identifier': instance_identifier} if instance_identifier else {}),
+                    'name': f'{instance_name} API Server',
                     # TODO: version the API
-                    'version': '0.1.0',
+                    'version': importlib.metadata.version('dandiapi'),
```

Comment:
this one though indeed would be "an improvement" so let's keep the library version here.